Libraries/pbxbuild/Sources/Tool/SwiftResolver.cpp

```diff
@@ -232,6 +232,73 @@ AppendObjcHeader(
     }
 }
 
+static void
+AppendUnderlyingModule(
+    std::vector<std::string> *args,
+    std::vector<Tool::Invocation::AuxiliaryFile> *auxiliaryFiles,
+    Tool::Context const *toolContext,
+    pbxsetting::Environment const &environment,
+    std::string const &headerName)
+{
+    /* No module map means no underlying module to import. */
+    ext::optional<Tool::ModuleMapInfo::Entry> const &moduleMap = toolContext->moduleMapInfo().moduleMap();
+    if (!moduleMap) {
+        return;
+    }
+
+    /*
+     * Create a module map to import the underlying module. However, exclude the Swift-generated
```

Comment:
This entire commit is so Swift libraries can use the ObjC version of their APIs internally, right?

Given your comment about avoiding importing the module's own structures, I'm guessing that's a bad thing; why is it not necessary?